stbt_rig.py

```diff
@@ -1,89 +1,197 @@
+
+            JENKINS INTEGRATION:
+              blah blah
+
+                  ./stbt_rig.py \\
+                    --node-id=stb-tester-00044b80ebeb \\
+                    run tests/roku.py::test_entering_the_settings_menu \\
+                        tests/roku.py::test_launching_iplayer_content
 
-    # Shouldn't pass auth token on command line because it would be visible in
+        """))
+
+    parser.add_argument(
+        "--portal-url", metavar="https://COMPANYNAME.stb-tester.com",
+        help="""Base URL of your Stb-tester Portal. You can specify it on the
+        command line or as "portal_url" in the [test_pack] section of
+        .stbt.conf. We look for .stbt.conf in the current working
+        directory.""")
+
+    # Can't pass auth token on command line because it would be visible in
     # /proc. Pass it in a file instead.
     parser.add_argument(
-        "--portal-auth-file",
-        help="name of file containing the HTTP REST API authentication token")
+        "--portal-auth-file", metavar="FILENAME",
+        help="""File containing the HTTP REST API authentication token.
+        See the AUTHENTICATION section below.""")
+
+    parser.add_argument(
+        "--node-id", metavar="stb-tester-abcdef123456",
+        help="""Which Stb-tester node to execute the COMMAND on. The node ID is
+        labelled on the physical Stb-tester node, and it is also shown in the
+        Stb-tester Portal.""")
 
-    parser.add_argument("--node-id")
-    parser.add_argument("--git-remote", default="origin")
+    parser.add_argument(
+        "--git-remote", metavar="NAME", default="origin",
+        help="""Which git remote to push to. Defaults to "origin" (this is the
+        default name that git creates when you did the original "git clone" of
+        the test-pack repository). This is only used by the commands that need
+        to push temporary snapshots to git: that is, "run" and "press" when
+        "--mode=interactive".""")
 
-    parser.add_argument("-v", "--verbose", action="count", dest="verbosity")
+    parser.add_argument(
+        "--mode", choices=["auto", "interactive", "jenkins"], default="auto",
+        help="""See the sections INTERACTIVE MODE and JENKINS INTEGRATION
+        below. This defaults to "auto", which detects whether or not it is
+        being run inside Jenkins.""")
+
+    parser.add_argument(
+        "-v", "--verbose", action="count", dest="verbosity", default=0,
+        help="""Specify once to enable INFO logging, twice for DEBUG.""")
 
-    sub = parser.add_subparsers(dest="command")
-    run_parser = sub.add_parser('run', help="Run a test-case")
+    subcommands = parser.add_subparsers(
+        dest="command", title="COMMANDS", metavar="COMMAND",
+        description=dedent("""\
+            Note: Run "./stbt_rig.py COMMAND -h" to see the additional
+            parameters for each COMMAND."""))
+
+    run_parser = subcommands.add_parser("run", help="Run test-cases")
```

Comment:
Perhaps we should refer to the `run_tests` REST API documentation here as a useful cross-reference.